src/main/java/xbdd/webapp/resource/feature/Stats.java

```diff
@@ -60,6 +62,49 @@ private DBObject getNumberOfAllStates(final DBCollection collection, final DBObj
 		return ret;
 	}
 
+	private static BasicDBObject getNumberOfAllStatesPerScenario(final BasicDBList features, final boolean separateManualAndAutomated) {
+		final String manualTag = "@manual";
+		final List<String> states = Arrays.asList("passed", "failed", "skipped", "undefined");
+		final BasicDBObject counts = new BasicDBObject(); // if `separateManualAndAutomated`, `counts` does not include manual scenarios
+		final BasicDBObject manualCounts = new BasicDBObject(); // if _not_ `separateManualAndAutomated`, `manualCounts` is not used
+		for (final String state : states) {
+			counts.append(state.substring(0, 1), 0);
+			manualCounts.append(state.substring(0, 1), 0);
+		}
+
+		for (final Object ob : features) {
+			final BasicDBList scenarios = (BasicDBList) ((DBObject) ob).get("elements");
+			if (scenarios != null) {
+				for (final Object o : scenarios) {
+					boolean isManual = false;
+					if (separateManualAndAutomated) {
+						// determine whether this scenario is marked @manual
```

Comment:
Possibly extract this to a `utils` class?

`ScenarioUtils.doesScenarioContainTag()`

Just a suggestion, trying to find a way to make the triple for loop look a bit nicer